PlayCover/Views/Sidebar Views/AppFolderView.swift

```diff
@@ -0,0 +1,272 @@
+                return false
+            } else if downloadVM.inProgress {
+                Log.shared.error(PlayCoverError.waitDownload)
+                return false
+            } else if let item = items.first {
+                if let identifier = item.registeredTypeIdentifiers.first {
+                    if identifier == "public.url" || identifier == "public.file-url" {
+                        item.loadItem(forTypeIdentifier: identifier, options: nil) { (urlData, _) in
+                            Task { @MainActor in
+                                if let urlData = urlData as? Data {
+                                    let url = NSURL(absoluteURLWithDataRepresentation: urlData, relativeTo: nil) as URL
+                                    if url.pathExtension == "ipa" {
+                                        installApp(url)
+                                    } else {
+                                        showWrongfileTypeAlert = true
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                return true
+            } else {
+                return false
+            }
+        }
+        .alert(isPresented: $showWrongfileTypeAlert) {
+            Alert(title: Text("alert.wrongFileType.title"),
+                  message: Text("alert.wrongFileType.subtitle"), dismissButton: .default(Text("button.OK")))
+        }
+        .alert("Legacy App Settings Detected!", isPresented: $showLegacyConvertAlert, actions: {
+            Button("button.Convert", role: .destructive) {
+                LegacySettings.convertLegacyMonolithPlist(LegacySettings.monolithURL)
+                do {
+                    try FileManager.default.removeItem(at: LegacySettings.monolithURL)
+                } catch {
+                    Log.shared.error(error)
+                }
+            }
+            .keyboardShortcut(.defaultAction)
+            Button("button.Cancel", role: .cancel) {
+                showLegacyConvertAlert.toggle()
+            }
+        }, message: {
+            Text("alert.legacyImport.subtitle")
+        })
+    }
+
+    private func installApp(_ url: URL) {
+        Installer.install(ipaUrl: url, export: false, returnCompletion: { _ in
+            Task { @MainActor in
+                appsVM.fetchApps()
+                NotifyService.shared.notify(
+                    NSLocalizedString("notification.appInstalled", comment: ""),
+                    NSLocalizedString("notification.appInstalled.message", comment: ""))
+            }
+        })
+    }
+
+    private func selectFile() {
+        NSOpenPanel.selectIPA { result in
+            if case .success(let url) = result {
+                installApp(url)
+            }
+        }
+    }
+}
+
+struct AddAppSheet: View {
+    @State var addSheetApps = false
+    @State var isAppEnabled: Bool
```

Comment:
`isAppEnabled` and `app` should be `Binding` instead of `State`.

Reply:
Oh ok, I thought I only needed to use `@Binding` when I need to "pass" the changed value to the other view and since I only use that value on `AddAppSheet` I didn't do that